nixos/unifi: Add the ability to specify extra JVM options

lucasbergman opened the pull request:
## Description of changes

The first two commits do some throat clearing, just cleaning up some formatting and making the command line args a list instead of a big string. The third commit gives me the ability to give the JVM zany options so I can watch its allocator and GC behavior, or monitor it with Prometheus using something like [jmx_exporter](https://github.com/prometheus/jmx_exporter).

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

lucasbergman committed:
nixos/unifi: Switch command line args to a list

There's no functionality change here, but this makes it a bit more
convenient to add more args.

lucasbergman committed:
nixos/unifi: Add extraJvmOptions option

This can be useful for adding extra logging, tweaking garbage collector
parameters, and for using JMX to monitor the JVM.

lucasbergman commented:
Hi @wegank by chance would you be interested in having a look at this? Forgive me if there's other protocol; I haven't committed to nixpkgs before. No rush.

wegank commented on nixos/modules/services/networking/unifi.nix:
```diff
@@ -21,16 +23,16 @@ in
       type = types.bool;
       default = false;
-      description = lib.mdDoc ''
+      description = mdDoc ''
```
It's actually the opposite that is recommended, see #208242.

lucasbergman replied:
That was a funny plot twist. I've now retconned the first commit (now 2bc02c0) to push in the right direction instead.

lucasbergman committed:
nixos/unifi: Clean up formatting

This removes "with lib", which advances #208242 a bit, and cleans up a few
formatting nits found by nixfmt.

lucasbergman commented:
Thanks very much for the review/approval. Does someone come by and merge this at some point? Sorry, I'm still new at this.

lucasbergman commented:
Hi @wegank or anyone else, do I need to do anything else for this to get merged?

wegank commented:
I'll merge it.

Would you also like to have a look at https://github.com/NixOS/nixpkgs/issues/195448#issuecomment-1785337014, as I assume you are an active UniFi user? It would be nice if we could upgrade to MongoDB 5.0 for the upcoming release.
